### ISSUE TITLE
Historical data of number/progress pulses

### ISSUE THREAD

**Issue by peterfication** [metabase/metabase#7198]
### Feature request

I have some questions in Metabase that return just a number (or progress). Now it would be nice to see how this number changes over time. However, my query is not able to produce time series data. That's why it would be awesome to have the ability to save the result of the query on a regular basis (eg. daily) to see the changes over time in a chart.

### Example

Let's say you have a database with blog posts. Now you want to see how many posts are public. This returns a number for the current time. This number might change as people hide or publish their posts. With this type of data, you are not able to query how many posts were public yesterday or draw a graph over time. (Please note, that this might be a very simplified example and that the query could be way more complex).

There are two possibilities right now:

1. Create a cron job yourself, that saves the result of this query every night.
2. Change the data model of your app to track publish/unpublish changes so you can query for the time series data.

Both approaches are not that nice as they involve development effort.

Do I miss a current feature of Metabase here or is there another easier way to achieve my goal without having the proposed feature?

Thanks a lot for answers!

**Comment by Shaddix** [comment]
+1 for that feature

**Comment by tlrobinson** [comment]
This would be neat. We'd need somewhere to store the data, so sort of related to #6094.

**Comment by bastians** [comment]
Sorry guys, but isn't that more a feature request for your own application? I mean, you can adjust your data model and then Metabase can do its job.

**Comment by peterfication** [comment]
I don't think really think so. Of course you can develop it in that way but you not always want something like that as part of your application code if it's just something BI needs for a short amount of time. And also, BI would need to wait for IT to prioritize it.

**Comment by bastians** [comment]
Well, obviously it means some development effort. For your team or for the Metabase team. From my point of view you should store this information with a timestamp in your database. Should cost that much ;-)

**Comment by ragboyjr** [comment]
I feel like this would be as simple as just having a simple pulse handler, that instead of slacking or emailing, just logs the metabase question + the pulse data, then we could easily build meta base reports off of the pulse log.

I'm not too familiar with the Metabase plugin system or if this type of customization is even supported by Metabase plugins, but wouldn't be a bad feature idea.

**Comment by adrianthomas** [comment]
> Well, obviously it means some development effort. For your team or for the Metabase team. From my point of view you should store this information with a timestamp in your database. Shouldn't cost that much ;-)

Part of the appeal of Metabase for us is that non-technical users can also easily build queries and information from multiple sources without requiring additional development or changes to the underlying data structures.

Being able to reference historical Metabase query results would be a really nice extension of those capabilities for those users. And for some 3rd party data sets, it would mean the difference of getting those values or not.

**Comment by maurociancio** [comment]
+1 to this feature. When you do not store historical data in your database, being able to run a query a let Metabase store the results, would be awesome.

**Comment by tbezemer** [comment]
+ 1 for this feature!! This could make Metabase a candidate for the data quality monitoring tool at the organisation I am currently working for (±5000 employees).

**Comment by econic** [comment]
I agree that this _could_ be solved by the inhouse devs of the application, but the big benefits of having this in metabase are to 

- allow non-tech personnel to achieve this in record time, even on suspicion ("might become helpful", which usually never gets prioritized in IT)
- allow this on data that is supposed to be transient (stats on a job queue, which should never maintain finished jobs)
- allow persisting these dashboards even if the underlying data isn't available anymore (aggregating data over all customers, surving events where customers got deleted for data protection reasons)

Since the functionality is pretty straightforward, I think the implementation on metabase's side should be manageable.
I'm thinking of this:

- enable logger for questions, asking the user to give the logger a name, let's "signups daily"
- invoking the question execution on a predefined schedule (interval / every day at 6pm / ...)
- the question result gets saved by metabase in its own data storage
- metabase automatically creates a model for that data, calls it "signups daily"

And from there on you can create a question about that model, e.g. plotting the recorded data on a graph.

**Comment by tobiashinz** [comment]
Any update on this?
